extensions/wikia/Email/Controller/WatchedPageController.class.php

```diff
@@ -26,8 +33,11 @@ public function initEmail() {
 
 		$this->title = \Title::newFromText( $titleText, $titleNamespace );
 		$this->summary = $this->getVal( 'summary' );
-		$this->currentRevId = $this->getVal( 'currentRevId' );
-		$this->previousRevId = $this->getVal( 'previousRevId' );
+
+		if ( $this->title instanceof \Title ) {
+			$this->currentRevId = $this->title->getLatestRevID( \Title::GAID_FOR_UPDATE );
+			$this->previousRevId = $this->title->getPreviousRevisionID( $this->currentRevId, \Title::GAID_FOR_UPDATE );
+		}
```

Comment:
```EmailNotifcation``` takes revIds from ```recentchanges``` table which doesn't store revIds for operations like ```delete``` ```move``` ```protect```, that's why we need to fetch them again here.